url/url.go

```diff
@@ -222,9 +224,19 @@ func (u *URL) fetchParams() {
 
 // ParseURL
 func Parse(inputURL string) (*URL, error) {
+	inputURL = decodeUnicodeEscapes(inputURL)
```

Comment:
I guess it's good to add this line inside `ParseURL` since `Parse` and `ParseURL` both exposed and `Parse` using `ParseURL`. what do you think?